extmod/cyw43_config_common.h

```diff
@@ -0,0 +1,143 @@
+#define CYW43_ETIMEDOUT                 MP_ETIMEDOUT // Connection timed out
+
+#define CYW43_THREAD_ENTER              MICROPY_PY_LWIP_ENTER
+#define CYW43_THREAD_EXIT               MICROPY_PY_LWIP_EXIT
+#define CYW43_THREAD_LOCK_CHECK
+
+#define CYW43_HOST_NAME                 mod_network_hostname_data
+
+#define CYW43_ARRAY_SIZE(a)             MP_ARRAY_SIZE(a)
+
+#define CYW43_HAL_PIN_MODE_INPUT        MP_HAL_PIN_MODE_INPUT
+#define CYW43_HAL_PIN_MODE_OUTPUT       MP_HAL_PIN_MODE_OUTPUT
+#define CYW43_HAL_PIN_PULL_NONE         MP_HAL_PIN_PULL_NONE
+#define CYW43_HAL_PIN_PULL_UP           MP_HAL_PIN_PULL_UP
+#define CYW43_HAL_PIN_PULL_DOWN         MP_HAL_PIN_PULL_DOWN
+
+#define CYW43_HAL_MAC_WLAN0             MP_HAL_MAC_WLAN0
+#define CYW43_HAL_MAC_BDADDR            MP_HAL_MAC_BDADDR
+
+#define cyw43_hal_ticks_us              mp_hal_ticks_us
+#define cyw43_hal_ticks_ms              mp_hal_ticks_ms
+
+#define cyw43_hal_pin_obj_t             mp_hal_pin_obj_t
+#define cyw43_hal_pin_config            mp_hal_pin_config
+#define cyw43_hal_pin_read              mp_hal_pin_read
+#define cyw43_hal_pin_low               mp_hal_pin_low
+#define cyw43_hal_pin_high              mp_hal_pin_high
+
+#define cyw43_hal_get_mac               mp_hal_get_mac
+#define cyw43_hal_get_mac_ascii         mp_hal_get_mac_ascii
+#define cyw43_hal_generate_laa_mac      mp_hal_generate_laa_mac
+
+#define cyw43_schedule_internal_poll_dispatch(func) pendsv_schedule_dispatch(PENDSV_DISPATCH_CYW43, func)
+
+// Note: this function is only called if CYW43_POST_POLL_HOOK is defined in cyw43_configport.h
+void cyw43_post_poll_hook(void);
+
+#ifdef MICROPY_EVENT_POLL_HOOK
+// Older style hook macros on some ports
+#define CYW43_EVENT_POLL_HOOK MICROPY_EVENT_POLL_HOOK
+#else
+// Newer style hooks on other ports
+#define CYW43_EVENT_POLL_HOOK mp_event_handle_nowait()
+#endif
+
+static inline void cyw43_delay_us(uint32_t us) {
+    uint32_t start = mp_hal_ticks_us();
+    while (mp_hal_ticks_us() - start < us) {
+    }
+}
+
+static inline void cyw43_delay_ms(uint32_t ms) {
+    // PendSV may be disabled via CYW43_THREAD_ENTER, so this delay is a busy loop.
+    uint32_t us = ms * 1000;
+    uint32_t start = mp_hal_ticks_us();
+    while (mp_hal_ticks_us() - start < us) {
+        CYW43_EVENT_POLL_HOOK;
+    }
+}
+
+#if LWIP_MDNS_RESPONDER == 1
+
+// Hook for any additional TCP/IP initialization than needs to be done.
+// Called after the netif specified by `itf` has been set up.
+#ifndef CYW43_CB_TCPIP_INIT_EXTRA
+#define CYW43_CB_TCPIP_INIT_EXTRA(self, itf) mdns_resp_add_netif(&self->netif[itf], mod_network_hostname_data)
+#endif
+
+// Hook for any additional TCP/IP deinitialization than needs to be done.
+// Called before the netif specified by `itf` is removed.
+#ifndef CYW43_CB_TCPIP_DEINIT_EXTRA
+#define CYW43_CB_TCPIP_DEINIT_EXTRA(self, itf) mdns_resp_remove_netif(&self->netif[itf])
+#endif
+
+#endif
+
+#if LWIP_MDNS_RESPONDER == 1
```

Comment:
This bit is added twice :joy:

Reply:
Oops, I must have messed up a rebase somewhere